packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -492,37 +494,63 @@ class OAuthClient extends CozyStackClient {
    * @returns {Promise} A promise that resolves with a new AccessToken object
    */
   async refreshToken() {
-    if (!this.isRegistered()) throw new NotRegisteredException()
-    if (!this.token) throw new Error('No token to refresh')
+    if (!this.isRegistered()) {
+      throw new NotRegisteredException()
+    }
+    if (!this.token) {
+      throw new Error('No token to refresh')
+    }
 
     const data = {
       grant_type: 'refresh_token',
       refresh_token: this.token.refreshToken,
       client_id: this.oauthOptions.clientID,
       client_secret: this.oauthOptions.clientSecret
     }
+    try {
+      const result = await this.fetchJSONWithCurrentToken(
+        'POST',
+        '/auth/access_token',
+        this.dataToQueryString(data),
+        {
+          headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+        }
+      )
 
-    const result = await super.fetchJSON(
-      'POST',
-      '/auth/access_token',
-      this.dataToQueryString(data),
-      {
-        headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+      const newToken = new AccessToken({
+        refresh_token: this.token.refreshToken,
+        ...result
+      })
+
+      this.setToken(newToken)
+
+      if (this.onTokenRefresh && typeof this.onTokenRefresh === 'function') {
+        this.onTokenRefresh(newToken)
       }
-    )
 
-    const newToken = new AccessToken({
-      refresh_token: this.token.refreshToken,
-      ...result
-    })
+      return newToken
+    } catch (e) {
+      if (errors.UNREGISTERED_CLIENT.test(e.message)) {
```

Comment:
Why not using `this.isRevocationError` ?

Reply:
Yes but then we need to update it because currently it only tests for `const CLIENT_NOT_FOUND = /Client not found/`